apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/AckError.cs

```diff
@@ -66,23 +60,31 @@ internal AckError(IEnumerable<KeyValuePair<string, string>> temporaryFailures, I
     internal IEnumerable<KeyValuePair<string, string>> PermanentFailures { get; }
 
     /// <summary>
-    /// Attempts to get the exception from the current instance of <seealso cref="AckError"/>.
+    /// Initializes a new instance of the <see cref="AckError"/> class.
     /// </summary>
-    /// <remarks>
-    /// This method will return <c>true</c> if and only if there are permanent failures
-    /// in the current instance of <seealso cref="AckError"/>,
-    /// in which case <paramref name="exception"/> will be populated.
-    /// </remarks>
-    /// <param name="exception">The <see cref="AggregateException"/> to populate if there are permanent failures.</param>
-    /// <returns><c>true</c> if there are permanent failures and <paramref name="exception"/> is populated, <c>false</c> otherwise.</returns>
-    internal bool TryGetAcknowledgementException(out AggregateException exception)
+    /// <param name="temporaryFailures">The collection of temporary failed AckId/NackId/ExtendId as key and their corresponding failure reason as value.</param>
+    /// <param name="permanentFailures">The collection of permanent failed AckId/NackId/ExtendId as key and their corresponding failure reason as value.</param>
+    internal AckError(IEnumerable<KeyValuePair<string, string>> temporaryFailures, IEnumerable<KeyValuePair<string, string>> permanentFailures)
     {
-        exception = HasPermanentFailures
-            ? new AggregateException(PermanentFailures.Select(AcknowledgementException.FromIdErrorPair).ToList())
-            : null;
-        return exception is not null;
+        TemporaryFailures = temporaryFailures;
+        PermanentFailures = permanentFailures;
     }
 
+    /// <summary>
+    /// Gets the permanent errors as the <see cref="IEnumerable{T}">IEnumerable</see>&lt;<see cref="AcknowledgementException"/>&gt;
+    /// from the current instance of <seealso cref="AckError"/>.
+    /// </summary>
+    /// <remarks>
+    /// This method will return the populated <see cref="IEnumerable{T}">IEnumerable</see>&lt;<see cref="AcknowledgementException"/>&gt; if there are permanent failures
```

Comment:
I think this would be simpler as:

```csharp
This method will always return a sequence of <see cref="AcknowledgementException"/>; this sequence will be empty if there are no permanent errors.
```

(Whatever we do, let's get rid of all the `&lt;` etc. It's hard to read here, and it won't help for browsing in VS really.)